### ISSUE TITLE
CI uses libccd 1.5, while the newest version is libccd 2.1

### ISSUE THREAD

**Issue by hongkai-dai** [flexible-collision-library/fcl#389]
Currently the CI uses libccd 1.5
![Screenshot from 2019-04-06 10-40-59](https://user-images.githubusercontent.com/6314000/55673153-c3642f80-5858-11e9-8841-63decc032cba.png)

The newest libccd is version 2.1

In my PR #388 , it fails all the tests in CI. The test would also fail on my local machine, if I build libccd 1.5. But when I build libccd 2.1 locally, all tests in that PR pass.

**Comment by SeanCurtis-TRI** [comment]
@jamiesnape Is there a good way to condition the test on a minimum version of libccd?

**Comment by sherm1** [comment]
Or should we modernize FCL CI to use libccd 2.1?

**Comment by SeanCurtis-TRI** [comment]
Certainly, unifying it would be a good thing. The homebrew for mac is using a different version than the apt-get on ubuntu Trusty (which is what is running the linux ci). Forcing all of them to use the same version would remove some of the noise there.

**Comment by scpeters** [comment]
Ubuntu Trusty is end-of-life this month (April 2019), so it might be a good time to make changes to the CI anyway

**Comment by SeanCurtis-TRI** [comment]
And with impeccable timing, Travis CI made [xenial available in November](https://blog.travis-ci.com/2018-11-08-xenial-release). Still, Xenial doesn't get us libccd 2.1. So, calling it out explicitly will probably give us the most stable results.

**Comment by wxmerkt** [comment]
I think this discussion links nicely to the one I opened upstream with ROS: https://github.com/ros/rosdistro/issues/20284. Are there any backports of libccd v2.1 currently being built and maintained or are you planning on doing so?

**Comment by jamiesnape** [comment]
I have a huge overhaul of the CI in the pipeline, including Bionic support. It is just a case of prioritizing it.

**Comment by jamiesnape** [comment]
(and if you want libccd 2.1 apt packages for CI or otherwise, building them is no trouble and we have the infrastructure to host them.)

**Comment by wxmerkt** [comment]
> (and if you want libccd 2.1 apt packages for CI or otherwise, building them is no trouble and we have the infrastructure to host them.)

This would be wonderful, especially if they could also be imported into the ROS apt.

**Comment by jamiesnape** [comment]
> especially if they could also be imported into the ROS apt.

That is probably for the `libccd` author or ROS to handle, to be honest.

> @jamiesnape Is there a good way to condition the test on a minimum version of libccd?

At least for recent `libccd`, I think the `CCD_VERSION` variable is defined, so given the old version of CMake in use, try
```
if(CCD_VERSION VERSION_EQUAL 2.1 OR CCD_VERSION VERSION_GREATER 2.1)
  add_test(...)
endif()
```

**Comment by jamiesnape** [comment]
@SeanCurtis-TRI @sherm1 Where this fit in the list of priorities? Feels like it is as equally blocked by a release as the other infrastructure improvements. Ideally, we would not be building `libccd` packages for Trusty, for instance.

**Comment by sherm1** [comment]
@SeanCurtis-TRI @hongkai-dai what's the priority to update FCL CI to use libccd 2.1?

**Comment by SeanCurtis-TRI** [comment]
We've addressed it to the degree that things aren't failing CI. Ideally, it would be good. But until we prioritize capping a release and pulling the trigger on the CI overhaul, I don't see there's really an practical way to advance this. I concur with @jamiesnape's assessment. So, I'd rephrase the question:

What's the priority of doing the CI overhaul?

**Comment by sherm1** [comment]
If it isn't blocking us at the moment I think it can wait until we've settled other FCL issues.
